caddy/module.go

```diff
@@ -69,6 +69,26 @@ func (f *FrankenPHPModule) Provision(ctx caddy.Context) error {
 		return fmt.Errorf(`expected ctx.App("frankenphp") to return *FrankenPHPApp, got nil`)
 	}
 
+	for i, wc := range f.Workers {
+		// Inherit environment variables from the parent php_server directive
+		if !filepath.IsAbs(wc.FileName) && f.Root != "" {
+			wc.FileName = filepath.Join(f.Root, wc.FileName)
+		}
+
+		if f.Env != nil {
+			if wc.Env == nil {
+				wc.Env = make(map[string]string)
+			}
+			for k, v := range f.Env {
+				// Only set if not already defined in the worker
+				if _, exists := wc.Env[k]; !exists {
+					wc.Env[k] = v
+				}
+			}
+		}
```

Comment:
You can probably make this a bit clearer (one of those few times I'd ever admit goto as being clearer, short of abstracting it to a function)

```go
if f.Env == nil {
    goto skipEnv;
}
if wc.Env == nil {
    wc.Env = make(map[string]string)
}
for k, v := range f.Env {
    if _, exists := wc.Env[k]; !exists {
        wc.Env[k] = v
    }
}
skipEnv:
```